Caltech-IPAC/firefly
# Firefly-1045: Fixed several bugs

#### [FIREFLY-1045](https://jira.ipac.caltech.edu/browse/FIREFLY-1045): Fix several bugs

- Fixed: Expanding/restoring an image in MultiProductViewer causes a reload stretch and color are reset
- Fixed: Exceptions happening when FItsHeaderView is up
- Fixed: Exceptions happening with StretchDropDownView
- Fixed: draw layer panel is hidden at times (zIndex was not correct)
- Fixed: on resize or when popping up mouse readout displayed zoomed to fix when it should not have
- Fixed: Moc validation was not accepting a valid FITs TFORM header
- Fixed: Better support for authorization in URLDownload.java
- Fixed: Expose MOC in Target Hips Panel
- Fixed: double load of web api
- Fixed: TAP search api did not init table correctly, also- fixed some bugs in examples
- Stretch should be sticky on MultiProductViewer
- Stretch should be stick for multi HDU images, if the same extension type
- [Firefly-1048](https://jira.ipac.caltech.edu/browse/FIREFLY-1049): Fixed: tooltip


#### Testing
 - https://fireflydev.ipac.caltech.edu/firefly-1045-bugs/firefly/
 - Some issues are too hard to test
 - Do an obscore search with TAP. (this test same bug found in wise, ztf, sofia)
     - MAST - ivoa - ivoa.obscore - m1  
    - change color on an image and expand then colapse - color should not change
    - change stretch and change image (by click on another row), stretch type should be sticky
 - on the tap panel - bring up visual target chooser - show layer dialog, you should be able to see it.
 - show a HiPS image and zoom in, resize the window and the HiPs will not rezoom, it should just stay the same
 - If you have an multi-hdu image with the same EXTTYPE keyword, upload, then changed the stretch, it will apply to all with that extension type
 - load https://fireflydev.ipac.caltech.edu/firefly-1045-bugs/firefly/?api=hips then click one of the the examples.  There should just be one HiPS loaded (not 2, fixes the double load)
